httpx/concurrency.py

```diff
@@ -225,6 +225,45 @@ def loop(self) -> asyncio.AbstractEventLoop:
 
         return reader, writer, protocol
 
+    async def start_tls(
```

Comment:
This is where a lot of the complexity of the functionality comes from. Can we verify that this is possible in a similar way for Trio cc @florimondmanca

Reply:
Will need to review this. :) Overall I think trio is very well equipped to deal with I/O situations. So once we get a working asyncio implementation in we’ll mostly have to see how to abstract it away in terms of concurrency backend. What do you think?

Reply:
Looking at it a bit more, I’ve already used trio TLS when implementing connect() in #214, but this seems to be something else. Can you confirm what this does is upgrade the existing TCP connection to start using TLS? What do you think of the web proxy example listed here? https://trio.readthedocs.io/en/stable/reference-io.html?highlight=TLS%20upgrade#the-abstract-stream-api

Reply:
Yeah this is more than just TLS on a new connection. `.connect()` is for new connections this is for adding TLS to a TCP stream, regardless of whether that TCP stream already has TLS on or not (since this is what occurs when you have an HTTPS proxy and you're connecting to an HTTPS service)